app/init.php

```diff
@@ -791,205 +233,172 @@ function (mixed $value) {
-            $dsn = $dsn[1] ?? '';
-            $config[] = $name;
-            if (empty($dsn)) {
-                //throw new Exception(Exception::GENERAL_SERVER_ERROR, "Missing value for DSN connection in {$key}");
-                continue;
-            }
+$registry->set(
+    'pools',
+    (function () {
+        $fallbackForDB = 'db_main=' . URL::unparse([
+            'scheme' => 'mariadb',
+            'host' => System::getEnv('_APP_DB_HOST', 'mariadb'),
+            'port' => System::getEnv('_APP_DB_PORT', '3306'),
+            'user' => System::getEnv('_APP_DB_USER', ''),
+            'pass' => System::getEnv('_APP_DB_PASS', ''),
+            'path' => System::getEnv('_APP_DB_SCHEMA', ''),
+        ]);
+        $fallbackForRedis = 'redis_main=' . URL::unparse([
+            'scheme' => 'redis',
+            'host' => System::getEnv('_APP_REDIS_HOST', 'redis'),
+            'port' => System::getEnv('_APP_REDIS_PORT', '6379'),
+            'user' => System::getEnv('_APP_REDIS_USER', ''),
+            'pass' => System::getEnv('_APP_REDIS_PASS', ''),
+        ]);
 
-            $dsn = new DSN($dsn);
-            $dsnHost = $dsn->getHost();
-            $dsnPort = $dsn->getPort();
-            $dsnUser = $dsn->getUser();
-            $dsnPass = $dsn->getPassword();
-            $dsnScheme = $dsn->getScheme();
-            $dsnDatabase = $dsn->getPath();
+        $connections = [
+            'console' => [
+                'type' => 'database',
+                'dsns' => System::getEnv('_APP_CONNECTIONS_DB_CONSOLE', $fallbackForDB),
+                'multiple' => false,
+                'schemes' => ['mariadb', 'mysql'],
+            ],
+            'database' => [
+                'type' => 'database',
+                'dsns' => System::getEnv('_APP_CONNECTIONS_DB_PROJECT', $fallbackForDB),
+                'multiple' => true,
+                'schemes' => ['mariadb', 'mysql'],
+            ],
+            'queue' => [
+                'type' => 'queue',
+                'dsns' => System::getEnv('_APP_CONNECTIONS_QUEUE', $fallbackForRedis),
+                'multiple' => false,
+                'schemes' => ['redis'],
+            ],
+            'pubsub' => [
+                'type' => 'pubsub',
+                'dsns' => System::getEnv('_APP_CONNECTIONS_PUBSUB', $fallbackForRedis),
+                'multiple' => false,
+                'schemes' => ['redis'],
+            ],
+            'cache' => [
+                'type' => 'cache',
+                'dsns' => System::getEnv('_APP_CONNECTIONS_CACHE', $fallbackForRedis),
+                'multiple' => true,
+                'schemes' => ['redis'],
+            ],
+        ];
 
-            if (!in_array($dsnScheme, $schemes)) {
-                throw new Exception(Exception::GENERAL_SERVER_ERROR, "Invalid console database scheme");
-            }
+        $pools = [];
+        $poolSize = (int)System::getEnv('_APP_POOL_SIZE', 64);
```

Comment:
lets make sure we also document this in the variables .php file and deprecate the old variables. 
And update the docker compose and docker compose.phtml files to use the new POOL_SIZE variable